ophyd/areadetector/plugins.py

```diff
@@ -106,6 +112,15 @@ def array_pixels(self):
 
         return pixels
 
+    def read_configuration(self):
+        ret = super().read_configuration()
+
+        source_port = self.nd_array_port.get()
+        source_plugin = self.root.get_plugin_by_port(source_port)
+        ret[source_port] = source_plugin.read_configuration()
```

Comment:
I like this idea, but won't this turn into:
{'port': dict(attr=value), ...}
where we actually want just a single merged dictionary?